test-frame-metrics-collector/src/test/java/io/skodjob/testframe/MetricsCollectorIT.java

```diff
@@ -0,0 +1,212 @@
+package io.skodjob.testframe;
+
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.PodList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.PodResource;
+import io.skodjob.testframe.clients.cmdClient.KubeCmdClient;
+import io.skodjob.testframe.clients.cmdClient.Kubectl;
+import io.skodjob.testframe.exceptions.IncompleteMetricsException;
+import io.skodjob.testframe.exceptions.MetricsCollectionException;
+import io.skodjob.testframe.exceptions.NoPodsFoundException;
+import io.skodjob.testframe.executor.Exec;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class MetricsCollectorIT {
```

Comment:
it seems to me more like unit tests instead of integration, but only my opinion

Reply:
I can rename it. Maybe `MetricsCollectorMock` or something similar?

Reply:
MetricsCollectorTest ?

Reply:
That test class already exist, and verifies the basic stuff without mocking external dependencies (i.e., exec, kubeClient, kubeCmdClient).

Reply:
ok ok my bad, in this case leave it as is

Reply:
> MetricsCollectorTest ?

It seems the same name is already used or?